### ISSUE TITLE
Traceback (but no negative effect) when botting through Barlett1932

### ISSUE THREAD

**Issue by suchow** [Dallinger/Dallinger#546]
When running through Bartlett1932 with a bot on `development`, I see the following traceback, though the experiment runs as expected and everything appears to correctly get stored in the database.

```
12:55:59 AM web.1   |  ERROR:dallinger.db:Exception during scoped worker transaction, rolling back.
12:55:59 AM web.1   |  Traceback (most recent call last):
12:55:59 AM web.1   |    File "/Users/jordansuchow/Dropbox/Berkeley/Projects/Current/Dallinger/dallinger/db.py", line 34, in sessions_scope
12:55:59 AM web.1   |      yield local_session
12:55:59 AM web.1   |    File "/Users/jordansuchow/Dropbox/Berkeley/Projects/Current/Dallinger/dallinger/db.py", line 64, in wrapper
12:55:59 AM web.1   |      return func(*args, **kwargs)
12:55:59 AM web.1   |    File "/Users/jordansuchow/Dropbox/Berkeley/Projects/Current/Dallinger/dallinger/experiment_server/experiment_server.py", line 1460, in worker_function
12:55:59 AM web.1   |      session.commit()
12:55:59 AM web.1   |    File "/Users/jordansuchow/.virtualenvs/dallinger/lib/python2.7/site-packages/sqlalchemy/orm/scoping.py", line 157, in do
12:55:59 AM web.1   |      return getattr(self.registry(), name)(*args, **kwargs)
12:55:59 AM web.1   |    File "/Users/jordansuchow/.virtualenvs/dallinger/lib/python2.7/site-packages/sqlalchemy/orm/session.py", line 874, in commit
12:55:59 AM web.1   |      self.transaction.commit()
12:55:59 AM web.1   |    File "/Users/jordansuchow/.virtualenvs/dallinger/lib/python2.7/site-packages/sqlalchemy/orm/session.py", line 468, in commit
12:55:59 AM web.1   |      self.session.dispatch.after_commit(self.session)
12:55:59 AM web.1   |    File "/Users/jordansuchow/.virtualenvs/dallinger/lib/python2.7/site-packages/sqlalchemy/event/attr.py", line 218, in __call__
12:55:59 AM web.1   |      fn(*args, **kw)
12:55:59 AM web.1   |    File "/Users/jordansuchow/Dropbox/Berkeley/Projects/Current/Dallinger/dallinger/db.py", line 128, in after_commit
12:55:59 AM web.1   |      for channel, message in session.info['outbox']:
12:55:59 AM web.1   |  KeyError: 'outbox'
12:55:59 AM web.1   |  ERROR:dallinger.db:Exception during scoped worker transaction, rolling back.
12:55:59 AM web.1   |  Traceback (most recent call last):
12:55:59 AM web.1   |    File "/Users/jordansuchow/Dropbox/Berkeley/Projects/Current/Dallinger/dallinger/db.py", line 34, in sessions_scope
12:55:59 AM web.1   |      yield local_session
12:55:59 AM web.1   |    File "/Users/jordansuchow/Dropbox/Berkeley/Projects/Current/Dallinger/dallinger/db.py", line 64, in wrapper
12:55:59 AM web.1   |      return func(*args, **kwargs)
12:55:59 AM web.1   |    File "/Users/jordansuchow/Dropbox/Berkeley/Projects/Current/Dallinger/dallinger/experiment_server/experiment_server.py", line 1273, in worker_complete
12:55:59 AM web.1   |      event_type='BotAssignmentSubmitted',
12:55:59 AM web.1   |    File "/Users/jordansuchow/Dropbox/Berkeley/Projects/Current/Dallinger/dallinger/experiment_server/experiment_server.py", line 1232, in _debug_notify
12:55:59 AM web.1   |      return worker_function(event_type, assignment_id, participant_id)
12:55:59 AM web.1   |    File "/Users/jordansuchow/Dropbox/Berkeley/Projects/Current/Dallinger/dallinger/db.py", line 64, in wrapper
12:55:59 AM web.1   |      return func(*args, **kwargs)
12:55:59 AM web.1   |    File "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/contextlib.py", line 35, in __exit__
12:55:59 AM web.1   |      self.gen.throw(type, value, traceback)
12:55:59 AM web.1   |    File "/Users/jordansuchow/Dropbox/Berkeley/Projects/Current/Dallinger/dallinger/db.py", line 46, in sessions_scope
12:55:59 AM web.1   |      local_session.rollback()
12:55:59 AM web.1   |    File "/Users/jordansuchow/.virtualenvs/dallinger/lib/python2.7/site-packages/sqlalchemy/orm/scoping.py", line 157, in do
12:55:59 AM web.1   |      return getattr(self.registry(), name)(*args, **kwargs)
12:55:59 AM web.1   |    File "/Users/jordansuchow/.virtualenvs/dallinger/lib/python2.7/site-packages/sqlalchemy/orm/session.py", line 838, in rollback
12:55:59 AM web.1   |      self.transaction.rollback()
12:55:59 AM web.1   |    File "/Users/jordansuchow/.virtualenvs/dallinger/lib/python2.7/site-packages/sqlalchemy/orm/session.py", line 477, in rollback
12:55:59 AM web.1   |      self._assert_active(prepared_ok=True, rollback_ok=True)
12:55:59 AM web.1   |    File "/Users/jordansuchow/.virtualenvs/dallinger/lib/python2.7/site-packages/sqlalchemy/orm/session.py", line 258, in _assert_active
12:55:59 AM web.1   |      "This session is in 'committed' state; no further "
12:55:59 AM web.1   |  InvalidRequestError: This session is in 'committed' state; no further SQL can be emitted within this transaction.
12:55:59 AM web.1   |  ERROR:Experiment_Server:Exception on /worker_complete [GET]
12:55:59 AM web.1   |  Traceback (most recent call last):
12:55:59 AM web.1   |    File "/Users/jordansuchow/.virtualenvs/dallinger/lib/python2.7/site-packages/flask/app.py", line 1817, in wsgi_app
12:55:59 AM web.1   |      response = self.full_dispatch_request()
12:55:59 AM web.1   |    File "/Users/jordansuchow/.virtualenvs/dallinger/lib/python2.7/site-packages/flask/app.py", line 1477, in full_dispatch_request
12:55:59 AM web.1   |      rv = self.handle_user_exception(e)
12:55:59 AM web.1   |    File "/Users/jordansuchow/.virtualenvs/dallinger/lib/python2.7/site-packages/flask/app.py", line 1381, in handle_user_exception
12:55:59 AM web.1   |      reraise(exc_type, exc_value, tb)
12:55:59 AM web.1   |    File "/Users/jordansuchow/.virtualenvs/dallinger/lib/python2.7/site-packages/flask/app.py", line 1475, in full_dispatch_request
12:55:59 AM web.1   |      rv = self.dispatch_request()
12:55:59 AM web.1   |    File "/Users/jordansuchow/.virtualenvs/dallinger/lib/python2.7/site-packages/flask/app.py", line 1461, in dispatch_request
12:55:59 AM web.1   |      return self.view_functions[rule.endpoint](**req.view_args)
12:55:59 AM web.1   |    File "/Users/jordansuchow/Dropbox/Berkeley/Projects/Current/Dallinger/dallinger/db.py", line 64, in wrapper
12:55:59 AM web.1   |      return func(*args, **kwargs)
12:55:59 AM web.1   |    File "/Users/jordansuchow/Dropbox/Berkeley/Projects/Current/Dallinger/dallinger/experiment_server/experiment_server.py", line 1273, in worker_complete
12:55:59 AM web.1   |      event_type='BotAssignmentSubmitted',
12:55:59 AM web.1   |    File "/Users/jordansuchow/Dropbox/Berkeley/Projects/Current/Dallinger/dallinger/experiment_server/experiment_server.py", line 1232, in _debug_notify
12:55:59 AM web.1   |      return worker_function(event_type, assignment_id, participant_id)
12:55:59 AM web.1   |    File "/Users/jordansuchow/Dropbox/Berkeley/Projects/Current/Dallinger/dallinger/db.py", line 64, in wrapper
12:55:59 AM web.1   |      return func(*args, **kwargs)
12:55:59 AM web.1   |    File "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/contextlib.py", line 35, in __exit__
12:55:59 AM web.1   |      self.gen.throw(type, value, traceback)
12:55:59 AM web.1   |    File "/Users/jordansuchow/Dropbox/Berkeley/Projects/Current/Dallinger/dallinger/db.py", line 46, in sessions_scope
12:55:59 AM web.1   |      local_session.rollback()
12:55:59 AM web.1   |    File "/Users/jordansuchow/.virtualenvs/dallinger/lib/python2.7/site-packages/sqlalchemy/orm/scoping.py", line 157, in do
12:55:59 AM web.1   |      return getattr(self.registry(), name)(*args, **kwargs)
12:55:59 AM web.1   |    File "/Users/jordansuchow/.virtualenvs/dallinger/lib/python2.7/site-packages/sqlalchemy/orm/session.py", line 838, in rollback
12:55:59 AM web.1   |      self.transaction.rollback()
12:55:59 AM web.1   |    File "/Users/jordansuchow/.virtualenvs/dallinger/lib/python2.7/site-packages/sqlalchemy/orm/session.py", line 477, in rollback
12:55:59 AM web.1   |      self._assert_active(prepared_ok=True, rollback_ok=True)
12:55:59 AM web.1   |    File "/Users/jordansuchow/.virtualenvs/dallinger/lib/python2.7/site-packages/sqlalchemy/orm/session.py", line 258, in _assert_active
12:55:59 AM web.1   |      "This session is in 'committed' state; no further "
12:55:59 AM web.1   |  InvalidRequestError: This session is in 'committed' state; no further SQL can be emitted within this transaction.
12:55:59 AM web.1   |  INFO:geventwebsocket.handler:127.0.0.1 - - [2017-04-17 00:55:59] "GET /worker_complete?uniqueId=YKQYB3%3AKQGURY HTTP/1.1" 500 412 0.156698
12:55:59 AM web.1   |  INFO:/Users/jordansuchow/Dropbox/Berkeley/Projects/Current/Dallinger/dallinger/recruiters.pyc:Created job 182cde06-3470-423b-a300-6b6510f5ee08 for url http://0.0.0.0:5000/ad?assignmentId=YQGK94&hitId=OQR806&workerId=KMM50M&mode=sandbox.
12:55:59 AM web.1   |  ERROR:dallinger.db:Exception during scoped worker transaction, rolling back.
12:55:59 AM web.1   |  Traceback (most recent call last):
12:55:59 AM web.1   |    File "/Users/jordansuchow/Dropbox/Berkeley/Projects/Current/Dallinger/dallinger/db.py", line 34, in sessions_scope
12:55:59 AM web.1   |      yield local_session
12:55:59 AM web.1   |    File "/Users/jordansuchow/Dropbox/Berkeley/Projects/Current/Dallinger/dallinger/db.py", line 64, in wrapper
12:55:59 AM web.1   |      return func(*args, **kwargs)
12:55:59 AM web.1   |    File "/Users/jordansuchow/Dropbox/Berkeley/Projects/Current/Dallinger/dallinger/experiment_server/experiment_server.py", line 1460, in worker_function
12:55:59 AM web.1   |      session.commit()
12:55:59 AM web.1   |    File "/Users/jordansuchow/.virtualenvs/dallinger/lib/python2.7/site-packages/sqlalchemy/orm/scoping.py", line 157, in do
12:55:59 AM web.1   |      return getattr(self.registry(), name)(*args, **kwargs)
12:55:59 AM web.1   |    File "/Users/jordansuchow/.virtualenvs/dallinger/lib/python2.7/site-packages/sqlalchemy/orm/session.py", line 874, in commit
12:55:59 AM web.1   |      self.transaction.commit()
12:55:59 AM web.1   |    File "/Users/jordansuchow/.virtualenvs/dallinger/lib/python2.7/site-packages/sqlalchemy/orm/session.py", line 468, in commit
12:55:59 AM web.1   |      self.session.dispatch.after_commit(self.session)
12:55:59 AM web.1   |    File "/Users/jordansuchow/.virtualenvs/dallinger/lib/python2.7/site-packages/sqlalchemy/event/attr.py", line 218, in __call__
12:55:59 AM web.1   |      fn(*args, **kw)
12:55:59 AM web.1   |    File "/Users/jordansuchow/Dropbox/Berkeley/Projects/Current/Dallinger/dallinger/db.py", line 128, in after_commit
12:55:59 AM web.1   |      for channel, message in session.info['outbox']:
12:55:59 AM web.1   |  KeyError: 'outbox'
12:55:59 AM web.1   |  ERROR:dallinger.db:Exception during scoped worker transaction, rolling back.
12:55:59 AM web.1   |  Traceback (most recent call last):
12:55:59 AM web.1   |    File "/Users/jordansuchow/Dropbox/Berkeley/Projects/Current/Dallinger/dallinger/db.py", line 34, in sessions_scope
12:55:59 AM web.1   |      yield local_session
12:55:59 AM web.1   |    File "/Users/jordansuchow/Dropbox/Berkeley/Projects/Current/Dallinger/dallinger/db.py", line 64, in wrapper
12:55:59 AM web.1   |      return func(*args, **kwargs)
12:55:59 AM web.1   |    File "/Users/jordansuchow/Dropbox/Berkeley/Projects/Current/Dallinger/dallinger/experiment_server/experiment_server.py", line 1273, in worker_complete
12:55:59 AM web.1   |      event_type='BotAssignmentSubmitted',
12:55:59 AM web.1   |    File "/Users/jordansuchow/Dropbox/Berkeley/Projects/Current/Dallinger/dallinger/experiment_server/experiment_server.py", line 1232, in _debug_notify
12:55:59 AM web.1   |      return worker_function(event_type, assignment_id, participant_id)
12:55:59 AM web.1   |    File "/Users/jordansuchow/Dropbox/Berkeley/Projects/Current/Dallinger/dallinger/db.py", line 64, in wrapper
12:55:59 AM web.1   |      return func(*args, **kwargs)
12:55:59 AM web.1   |    File "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/contextlib.py", line 35, in __exit__
12:55:59 AM web.1   |      self.gen.throw(type, value, traceback)
12:55:59 AM web.1   |    File "/Users/jordansuchow/Dropbox/Berkeley/Projects/Current/Dallinger/dallinger/db.py", line 46, in sessions_scope
12:55:59 AM web.1   |      local_session.rollback()
12:55:59 AM web.1   |    File "/Users/jordansuchow/.virtualenvs/dallinger/lib/python2.7/site-packages/sqlalchemy/orm/scoping.py", line 157, in do
12:55:59 AM web.1   |      return getattr(self.registry(), name)(*args, **kwargs)
12:55:59 AM web.1   |    File "/Users/jordansuchow/.virtualenvs/dallinger/lib/python2.7/site-packages/sqlalchemy/orm/session.py", line 838, in rollback
12:55:59 AM web.1   |      self.transaction.rollback()
12:55:59 AM web.1   |    File "/Users/jordansuchow/.virtualenvs/dallinger/lib/python2.7/site-packages/sqlalchemy/orm/session.py", line 477, in rollback
12:55:59 AM web.1   |      self._assert_active(prepared_ok=True, rollback_ok=True)
12:55:59 AM web.1   |    File "/Users/jordansuchow/.virtualenvs/dallinger/lib/python2.7/site-packages/sqlalchemy/orm/session.py", line 258, in _assert_active
12:55:59 AM web.1   |      "This session is in 'committed' state; no further "
12:55:59 AM web.1   |  InvalidRequestError: This session is in 'committed' state; no further SQL can be emitted within this transaction.
12:55:59 AM web.1   |  ERROR:Experiment_Server:Exception on /worker_complete [GET]
12:55:59 AM web.1   |  Traceback (most recent call last):
12:55:59 AM web.1   |    File "/Users/jordansuchow/.virtualenvs/dallinger/lib/python2.7/site-packages/flask/app.py", line 1817, in wsgi_app
12:55:59 AM web.1   |      response = self.full_dispatch_request()
12:55:59 AM web.1   |    File "/Users/jordansuchow/.virtualenvs/dallinger/lib/python2.7/site-packages/flask/app.py", line 1477, in full_dispatch_request
12:55:59 AM web.1   |      rv = self.handle_user_exception(e)
12:55:59 AM web.1   |    File "/Users/jordansuchow/.virtualenvs/dallinger/lib/python2.7/site-packages/flask/app.py", line 1381, in handle_user_exception
12:55:59 AM web.1   |      reraise(exc_type, exc_value, tb)
12:55:59 AM web.1   |    File "/Users/jordansuchow/.virtualenvs/dallinger/lib/python2.7/site-packages/flask/app.py", line 1475, in full_dispatch_request
12:55:59 AM web.1   |      rv = self.dispatch_request()
12:55:59 AM web.1   |    File "/Users/jordansuchow/.virtualenvs/dallinger/lib/python2.7/site-packages/flask/app.py", line 1461, in dispatch_request
12:55:59 AM web.1   |      return self.view_functions[rule.endpoint](**req.view_args)
12:55:59 AM web.1   |    File "/Users/jordansuchow/Dropbox/Berkeley/Projects/Current/Dallinger/dallinger/db.py", line 64, in wrapper
12:55:59 AM web.1   |      return func(*args, **kwargs)
12:55:59 AM web.1   |    File "/Users/jordansuchow/Dropbox/Berkeley/Projects/Current/Dallinger/dallinger/experiment_server/experiment_server.py", line 1273, in worker_complete
12:55:59 AM web.1   |      event_type='BotAssignmentSubmitted',
12:55:59 AM web.1   |    File "/Users/jordansuchow/Dropbox/Berkeley/Projects/Current/Dallinger/dallinger/experiment_server/experiment_server.py", line 1232, in _debug_notify
12:55:59 AM web.1   |      return worker_function(event_type, assignment_id, participant_id)
12:55:59 AM web.1   |    File "/Users/jordansuchow/Dropbox/Berkeley/Projects/Current/Dallinger/dallinger/db.py", line 64, in wrapper
12:55:59 AM web.1   |      return func(*args, **kwargs)
12:55:59 AM web.1   |    File "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/contextlib.py", line 35, in __exit__
12:55:59 AM web.1   |      self.gen.throw(type, value, traceback)
12:55:59 AM web.1   |    File "/Users/jordansuchow/Dropbox/Berkeley/Projects/Current/Dallinger/dallinger/db.py", line 46, in sessions_scope
12:55:59 AM web.1   |      local_session.rollback()
12:55:59 AM web.1   |    File "/Users/jordansuchow/.virtualenvs/dallinger/lib/python2.7/site-packages/sqlalchemy/orm/scoping.py", line 157, in do
12:55:59 AM web.1   |      return getattr(self.registry(), name)(*args, **kwargs)
12:55:59 AM web.1   |    File "/Users/jordansuchow/.virtualenvs/dallinger/lib/python2.7/site-packages/sqlalchemy/orm/session.py", line 838, in rollback
12:55:59 AM web.1   |      self.transaction.rollback()
12:55:59 AM web.1   |    File "/Users/jordansuchow/.virtualenvs/dallinger/lib/python2.7/site-packages/sqlalchemy/orm/session.py", line 477, in rollback
12:55:59 AM web.1   |      self._assert_active(prepared_ok=True, rollback_ok=True)
12:55:59 AM web.1   |    File "/Users/jordansuchow/.virtualenvs/dallinger/lib/python2.7/site-packages/sqlalchemy/orm/session.py", line 258, in _assert_active
12:55:59 AM web.1   |      "This session is in 'committed' state; no further "
12:55:59 AM web.1   |  InvalidRequestError: This session is in 'committed' state; no further SQL can be emitted within this transaction.
12:55:59 AM web.1   |  INFO:geventwebsocket.handler:127.0.0.1 - - [2017-04-17 00:55:59] "GET /worker_complete?uniqueId=YKQYB3%3AKQGURY HTTP/1.1" 500 412 0.098881
12:55:59 AM web.1   |  INFO:/Users/jordansuchow/Dropbox/Berkeley/Projects/Current/Dallinger/dallinger/recruiters.pyc:Created job bef8bcae-8f84-4854-86cb-a5744fa505da for url http://0.0.0.0:5000/ad?assignmentId=IT3VBA&hitId=1D1FJJ&workerId=EIJBUN&mode=sandbox.
12:55:59 AM web.1   |  ERROR:dallinger.db:Exception during scoped worker transaction, rolling back.
12:55:59 AM web.1   |  Traceback (most recent call last):
12:55:59 AM web.1   |    File "/Users/jordansuchow/Dropbox/Berkeley/Projects/Current/Dallinger/dallinger/db.py", line 34, in sessions_scope
12:55:59 AM web.1   |      yield local_session
12:55:59 AM web.1   |    File "/Users/jordansuchow/Dropbox/Berkeley/Projects/Current/Dallinger/dallinger/db.py", line 64, in wrapper
12:55:59 AM web.1   |      return func(*args, **kwargs)
12:55:59 AM web.1   |    File "/Users/jordansuchow/Dropbox/Berkeley/Projects/Current/Dallinger/dallinger/experiment_server/experiment_server.py", line 1460, in worker_function
12:55:59 AM web.1   |      session.commit()
12:55:59 AM web.1   |    File "/Users/jordansuchow/.virtualenvs/dallinger/lib/python2.7/site-packages/sqlalchemy/orm/scoping.py", line 157, in do
12:55:59 AM web.1   |      return getattr(self.registry(), name)(*args, **kwargs)
12:55:59 AM web.1   |    File "/Users/jordansuchow/.virtualenvs/dallinger/lib/python2.7/site-packages/sqlalchemy/orm/session.py", line 874, in commit
12:55:59 AM web.1   |      self.transaction.commit()
12:55:59 AM web.1   |    File "/Users/jordansuchow/.virtualenvs/dallinger/lib/python2.7/site-packages/sqlalchemy/orm/session.py", line 468, in commit
12:55:59 AM web.1   |      self.session.dispatch.after_commit(self.session)
12:55:59 AM web.1   |    File "/Users/jordansuchow/.virtualenvs/dallinger/lib/python2.7/site-packages/sqlalchemy/event/attr.py", line 218, in __call__
12:55:59 AM web.1   |      fn(*args, **kw)
12:55:59 AM web.1   |    File "/Users/jordansuchow/Dropbox/Berkeley/Projects/Current/Dallinger/dallinger/db.py", line 128, in after_commit
12:55:59 AM web.1   |      for channel, message in session.info['outbox']:
12:55:59 AM web.1   |  KeyError: 'outbox'
12:55:59 AM web.1   |  ERROR:dallinger.db:Exception during scoped worker transaction, rolling back.
12:55:59 AM web.1   |  Traceback (most recent call last):
12:55:59 AM web.1   |    File "/Users/jordansuchow/Dropbox/Berkeley/Projects/Current/Dallinger/dallinger/db.py", line 34, in sessions_scope
12:55:59 AM web.1   |      yield local_session
12:55:59 AM web.1   |    File "/Users/jordansuchow/Dropbox/Berkeley/Projects/Current/Dallinger/dallinger/db.py", line 64, in wrapper
12:55:59 AM web.1   |      return func(*args, **kwargs)
12:55:59 AM web.1   |    File "/Users/jordansuchow/Dropbox/Berkeley/Projects/Current/Dallinger/dallinger/experiment_server/experiment_server.py", line 1273, in worker_complete
12:55:59 AM web.1   |      event_type='BotAssignmentSubmitted',
12:55:59 AM web.1   |    File "/Users/jordansuchow/Dropbox/Berkeley/Projects/Current/Dallinger/dallinger/experiment_server/experiment_server.py", line 1232, in _debug_notify
12:55:59 AM web.1   |      return worker_function(event_type, assignment_id, participant_id)
12:55:59 AM web.1   |    File "/Users/jordansuchow/Dropbox/Berkeley/Projects/Current/Dallinger/dallinger/db.py", line 64, in wrapper
12:55:59 AM web.1   |      return func(*args, **kwargs)
12:55:59 AM web.1   |    File "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/contextlib.py", line 35, in __exit__
12:55:59 AM web.1   |      self.gen.throw(type, value, traceback)
12:55:59 AM web.1   |    File "/Users/jordansuchow/Dropbox/Berkeley/Projects/Current/Dallinger/dallinger/db.py", line 46, in sessions_scope
12:55:59 AM web.1   |      local_session.rollback()
12:55:59 AM web.1   |    File "/Users/jordansuchow/.virtualenvs/dallinger/lib/python2.7/site-packages/sqlalchemy/orm/scoping.py", line 157, in do
12:55:59 AM web.1   |      return getattr(self.registry(), name)(*args, **kwargs)
12:55:59 AM web.1   |    File "/Users/jordansuchow/.virtualenvs/dallinger/lib/python2.7/site-packages/sqlalchemy/orm/session.py", line 838, in rollback
12:55:59 AM web.1   |      self.transaction.rollback()
12:55:59 AM web.1   |    File "/Users/jordansuchow/.virtualenvs/dallinger/lib/python2.7/site-packages/sqlalchemy/orm/session.py", line 477, in rollback
12:55:59 AM web.1   |      self._assert_active(prepared_ok=True, rollback_ok=True)
12:55:59 AM web.1   |    File "/Users/jordansuchow/.virtualenvs/dallinger/lib/python2.7/site-packages/sqlalchemy/orm/session.py", line 258, in _assert_active
12:55:59 AM web.1   |      "This session is in 'committed' state; no further "
12:55:59 AM web.1   |  InvalidRequestError: This session is in 'committed' state; no furt
12:55:59 AM web.1   |  her SQL can be emitted within this transaction.
12:55:59 AM web.1   |  ERROR:Experiment_Server:Exception on /worker_complete [GET]
12:55:59 AM web.1   |  Traceback (most recent call last):
12:55:59 AM web.1   |    File "/Users/jordansuchow/.virtualenvs/dallinger/lib/python2.7/site-packages/flask/app.py", line 1817, in wsgi_app
12:55:59 AM web.1   |      response = self.full_dispatch_request()
12:55:59 AM web.1   |    File "/Users/jordansuchow/.virtualenvs/dallinger/lib/python2.7/site-packages/flask/app.py", line 1477, in full_dispatch_request
12:55:59 AM web.1   |      rv = self.handle_user_exception(e)
12:55:59 AM web.1   |    File "/Users/jordansuchow/.virtualenvs/dallinger/lib/python2.7/site-packages/flask/app.py", line 1381, in handle_user_exception
12:55:59 AM web.1   |      reraise(exc_type, exc_value, tb)
12:55:59 AM web.1   |    File "/Users/jordansuchow/.virtualenvs/dallinger/lib/python2.7/site-packages/flask/app.py", line 1475, in full_dispatch_request
12:55:59 AM web.1   |      rv = self.dispatch_request()
12:55:59 AM web.1   |    File "/Users/jordansuchow/.virtualenvs/dallinger/lib/python2.7/site-packages/flask/app.py", line 1461, in dispatch_request
12:55:59 AM web.1   |      return self.view_functions[rule.endpoint](**req.view_args)
12:55:59 AM web.1   |    File "/Users/jordansuchow/Dropbox/Berkeley/Projects/Current/Dallinger/dallinger/db.py", line 64, in wrapper
12:55:59 AM web.1   |      return func(*args, **kwargs)
12:55:59 AM web.1   |    File "/Users/jordansuchow/Dropbox/Berkeley/Projects/Current/Dallinger/dallinger/experiment_server/experiment_server.py", line 1273, in worker_complete
12:55:59 AM web.1   |      event_type='BotAssignmentSubmitted',
12:55:59 AM web.1   |    File "/Users/jordansuchow/Dropbox/Berkeley/Projects/Current/Dallinger/dallinger/experiment_server/experiment_server.py", line 1232, in _debug_notify
12:55:59 AM web.1   |      return worker_function(event_type, assignment_id, participant_id)
12:55:59 AM web.1   |    File "/Users/jordansuchow/Dropbox/Berkeley/Projects/Current/Dallinger/dallinger/db.py", line 64, in wrapper
12:55:59 AM web.1   |      return func(*args, **kwargs)
12:55:59 AM web.1   |    File "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/contextlib.py", line 35, in __exit__
12:55:59 AM web.1   |      self.gen.throw(type, value, traceback)
12:55:59 AM web.1   |    File "/Users/jordansuchow/Dropbox/Berkeley/Projects/Current/Dallinger/dallinger/db.py", line 46, in sessions_scope
12:55:59 AM web.1   |      local_session.rollback()
12:55:59 AM web.1   |    File "/Users/jordansuchow/.virtualenvs/dallinger/lib/python2.7/site-packages/sqlalchemy/orm/scoping.py", line 157, in do
12:55:59 AM web.1   |      return getattr(self.registry(), name)(*args, **kwargs)
12:55:59 AM web.1   |    File "/Users/jordansuchow/.virtualenvs/dallinger/lib/python2.7/site-packages/sqlalchemy/orm/session.py", line 838, in rollback
12:55:59 AM web.1   |      self.transaction.rollback()
12:55:59 AM web.1   |    File "/Users/jordansuchow/.virtualenvs/dallinger/lib/python2.7/site-packages/sqlalchemy/orm/session.py", line 477, in rollback
12:55:59 AM web.1   |      self._assert_active(prepared_ok=True, rollback_ok=True)
12:55:59 AM web.1   |    File "/Users/jordansuchow/.virtualenvs/dallinger/lib/python2.7/site-packages/sqlalchemy/orm/session.py", line 258, in _assert_active
12:55:59 AM web.1   |      "This session is in 'committed' state; no further "
12:55:59 AM web.1   |  InvalidRequestError: This session is in 'committed' state; no further SQL can be emitted within this transaction.
```

## Expected Behavior
No exceptions.

## Current Behavior
`InvalidRequestError`, no other problems noticed.

## Steps to Reproduce (for bugs)
1. In Bartlett1932, set:

```
recruiter = bots
webdriver_type = firefox
```

2. Run `dallinger debug --verbose`

**Comment by alecpm** [comment]
I committed a fix for the `KeyError: 'outbox'` issue on my `packaged-experiments` branch in f194047e53f56d0c3e85eda94cf8077dd372f389. I'll port it onto development.

**Comment by alecpm** [comment]
Fixed in 186a98e6c0eb477db0c0c3f5436f3dcbb14bb7ba

**Comment by suchow** [comment]
@alecpm Sigh, how am I supposed to get issues ready for "testing and bug fixing" week if you're solving them within minutes of my posting them? ;)
